pkg/generate/app/pipeline.go

```diff
@@ -28,21 +35,76 @@ type Pipeline struct {
 	Labels     map[string]string
 }
 
-// NewImagePipeline creates a new pipeline with components that are not expected
-// to be built.
-func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
-	return &Pipeline{
-		From:  from,
-		Image: image,
-	}, nil
+// NewPipeline creates a new pipeline from a component reference.
+func NewPipeline(name string, names map[string]int, ref ComponentReference, environment Environment, addEnvToBuild bool, labels map[string]string, outputDocker bool, deploy bool) (*Pipeline, error) {
```

Comment:
I guess it's fine for now, but we try to keep component reference separate
from lower level code.  Can you add a todo to the method definition to
refactor that?

On Nov 20, 2015, at 4:01 PM, Rodolfo Carvalho <notifications@github.com>
wrote:

In pkg/generate/app/pipeline.go
<https://github.com/openshift/origin/pull/5992#discussion_r45517943>:

> @@ -28,21 +35,76 @@ type Pipeline struct {
>  	Labels     map[string]string
>  }
>
> -// NewImagePipeline creates a new pipeline with components that are not expected
> -// to be built.
> -func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
> -	return &Pipeline{
> -		From:  from,
> -		Image: image,
> -	}, nil
> +// NewPipeline creates a new pipeline from a component reference.
> +func NewPipeline(name string, names map[string]int, ref ComponentReference, environment Environment, addEnvToBuild bool, labels map[string]string, outputDocker bool, deploy bool) (*Pipeline, error) {

@smarterclayton <https://github.com/smarterclayton> this is coming from
extracting the inner loop portion of buildPipelines
<https://github.com/openshift/origin/pull/5992/files#diff-67f42376bdba60c035c6268d160059baR585>
into a separate function.
I'm not familiar with the original design, though. Could you elaborate a
bit more on why you don't like it here?

We need that ref for various things, like calling InputImageFromMatch,
StrategyAndSourceForRepository, ...

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5992/files#r45517943>.